### ISSUE TITLE
[suggestion] don't flush queue / save resume queue

### ISSUE THREAD

**Issue by Leerz** [Cruel/freeShop#112]
Network issues and possible downloading bugs are sometimes random. maybe we can implement a queue saving method. :)

(Bugs in downloading, unreliable internet)
/Sos observed and happened like twice with more than 10 in queue, network went dead and boom.

Thanks

**Comment by Leerz** [comment]
Seems like this is already implemented, I'm unsure, does it work even after restart of the system? Twice did it happen that the queue is gone.

**Comment by Cruel** [comment]
Did this happen on latest version 2.0.4? And what happened when network went dead?

**Comment by Leerz** [comment]
Yes, on 2.0.4, there was nothing downloading after the network is dead, it didn't crash or anything, it just didn't resume the queue

(Basically i had 20+ in queue, 
After the 10th finished, (and after the network issue)) .
The previous says installed, then the next says queue
I should have taken a photo of it.

I closed freeshop and opened it again, was it supposed to save the previous' unfinished queue? queue

**Comment by Cruel** [comment]
It's supposed to save the queue. So if it doesn't, it's a bug and we got to figure out how to reproduce the problem. Also, it should be able to recover better on network failures, so that'd be a separate bug if it can be duplicated.

I know I've seen it fail to save the queue, but only when there is a crash or it freezes so it doesn't exit properly. And I've never seen the queue stop progressing, I think that is likely the cause.

I'll have to leave the issue open until more people experience the issue and can report how it happens.

**Comment by Leerz** [comment]
Thank you, i can't seem to replicate the problem, given that I've been testing and trying to recreate it, queue and downloading seem to be handled phenomenally. 

-edit- and found a crash.
10 in queue, allowed the first 3 to finishes, turned off my internet (wifi is on, but no data conn.) 
After several retries line said something like network problem/error.  Queue does try to cycle through the next in queue and retries, then network problem is in the line, 
I turn the net back on, it processes the last in queue, after it is finished, i hit the retry button, then it crashes :)
Turned the conn ba k o

**Comment by Cruel** [comment]
Thanks for all the testing. Reminds me there _are_ some bugs with the download restart button that I forgot to fix. I was just busy fixing the causes of failure in the first place, so I'll have to force a failure in code to test the restarting.

I also need to change the way it restarts with bad internet. When it happened for you, did it cycle between "Suspended" with occasional "Retrying... 1" with the number never changing?

**Comment by Leerz** [comment]
Can't answer for certain, but i recall cycling between queue Nd installed rows, no retrying (maybe i didn't noticed it?)
But I'm pretty positive * of the ff:
Test 1.0
1. It didn't crash the first time
1.1 it did not recover the cueue

Test 2.0
2.1 !!details posted above

**Comment by aoblue1** [comment]
since 2.0.4 i'm having a hell out of bugs like not downloading, shop closing even restarting the console, a lot of internet problems? errors and now the shop close and the icon on the menu its gone and i had to turn off and on again so it appear again
also a lots of "retryings" and i think its when enter in the sleeping mode i really dnt know

**Comment by Cruel** [comment]
Sounds like you have a slow/bad connection. You may need to wait for #83 so you can increase timeout value.

Atm, if you download ~20 KB/s or less, it'll probably screw up. I'll have the above changes done in a couple days.

**Comment by aoblue1** [comment]
ok i think thats the problem ty bro ill keep testing
